docs/src/modules/components/Demo.test.js

```diff
@@ -0,0 +1,42 @@
+import * as React from 'react';
+import Demo from './Demo';
+
+const { expect } = require('chai');
+const { createRenderer } = require('@mui/internal-test-utils');
+
+describe('Function execution time test', () => {
+  const { render } = createRenderer();
+  let str = '';
+  for (let i = 0; i < 1000000; i += 1) {
+    str += '.';
+  }
+  str += '\n@.ts';
+
+  it('should complete within 10 seconds', () => {
```

Comment:
Hey @SuperMaxine thanks for working on this, just wondering why `10 seconds`? At first glance it seems a bit arbitrary

Reply:
This is just a loose condition. In fact, if there is no ReDoS, the runtime of these tests in my local environment is around 0.0x seconds. However, if ReDoS exists (i.e., if my fixes in the source code are removed), these examples might hang for several minutes or even longer. Since I don’t have much information about the performance of the machines used for automated review checks, I chose an upper limit that would never be hit under normal situation. If convenient, please adjust this to a more practical value based on your actual needs.